### PR TITLE
assets should get served from cache

### PR DESCRIPTION
**Pull request by GithinjiHans**

### CODE REVIEW

File: src/service-worker/service-worker.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ import { clientsClaim } from "workbox-core";
 import { ExpirationPlugin } from "workbox-expiration";
 import { precacheAndRoute, createHandlerBoundToURL } from "workbox-precaching";
 import { registerRoute } from "workbox-routing";
-import { StaleWhileRevalidate } from "workbox-strategies";
+import { CacheFirst } from "workbox-strategies";
```
Comment (tijlleenders) [inline src/service-worker/service-worker.ts:16]: Why are we changing strategies?

Comment (GithinjiHans) [reply src/service-worker/service-worker.ts:16]: Assets are served from the cache storage using CacheFirst strategy....if they are not found in the cache, they are requested from the server then cached and the response from the server is served initially.

Comment (tijlleenders) [reply src/service-worker/service-worker.ts:16]: Thanks for the explanation - but how is this different from stale-while-revalidate?

Comment (pulkitvyas08) [reply src/service-worker/service-worker.ts:16]: While the implementation details under the hood are quite similar for both, what I observed is that the fonts and SVG are loading much faster in this version as compared to the zinzen.vercel.app. @tijlleenders Can you also do the same comparison on, say regular 2G, and check if that's true

Comment (tijlleenders) [reply src/service-worker/service-worker.ts:16]: @GithinjiHans Can you tell me what the theoretical difference is between the two caching strategies CacheFirst and StaleWhileRevalidate?

**Review by tijlleenders — CHANGES_REQUESTED** [review]
The font is not cached yet.

File: src/service-worker/service-worker.ts (hunk 1 above)
Comment (GithinjiHans) [reply src/service-worker/service-worker.ts:16]: Stalewhile-revalidate also serves assets from cache but the assets in the cache are checked if they're upto date, if not they are not upto date they're updated and the latest version is served mostly used in live apps. Thats the main difference between Stalewhile-revalidate and Cache first.

**Comment by GithinjiHans** [comment]
> The font is not cached yet.



> The font is not cached yet.

I am unable to produce the error on FF I'll correct it once am able to produce it on chrome

File: src/service-worker/service-worker.ts (hunk 1 above)
Comment (tijlleenders) [reply src/service-worker/service-worker.ts:16]: Ok - so we want [stale-while-revalidate](https://developer.chrome.com/docs/workbox/modules/workbox-strategies/#stale-while-revalidate) - not [cache-first](https://developer.chrome.com/docs/workbox/modules/workbox-strategies/#cache-first-cache-falling-back-to-network) as cache first will never update the cache.

If the cache is never updated then users will never get updates.
So unless you have an option to set an interval for checking/updating the cache we should use stale-while-revalidate.

**Comment by tijlleenders** [comment]
> > The font is not cached yet.
> 
> > The font is not cached yet.
> 
> I am unable to produce the error on FF I'll correct it once am able to produce it on chrome

Hint: the font is a .woff2 file and you have not configured anything to cache that - so it should be reproducible.
image

File: src/service-worker/service-worker.ts (hunk 1 above)
Comment (tijlleenders) [reply src/service-worker/service-worker.ts:16]: Page loads using stale-while-revalidate should be just as fast as page loads using cache-first, right?

Comment (tijlleenders) [inline src/service-worker/service-worker.ts:16]:
```suggestion
import { StaleWhileRevalidate} from "workbox-strategies";
```

**Comment by tijlleenders** [comment]
Even though CacheFirst is not faster than StaleWhileRevalidate it is preferred because it doesn't cause unnecessary network traffic, so it is:
- Cheaper
- Safer (no usage info 'leaking' to the network)

We don't have to build anything to refresh the cache: When a file changes the content hash in the filename will change => forcing a cache miss and load from network.

@GithinjiHans Thanks for changing it!

**Review by tijlleenders — DISMISSED** [review]

**Review by tijlleenders — APPROVED** [review]

**Pull request merged**